src/classes/PS_CommitmentRequest.cls

```diff
@@ -155,6 +155,7 @@ public inherited sharing class PS_CommitmentRequest {
             : UserInfo.getDefaultCurrency();
 
         reqBody.withCurrency(currencyCode);
+        reqBody.withMetadata(new Metadata());
```

Comment:
Would it be better for the metadata to be in the buildCreateRequest method?  It seems more suited to the sort of data being built there.

Reply:
That is a good suggestion, I thought of changing this so it is set as a default, but it will not work when campaign code is to be passed. Since fields such as gateway, donor info, currency, etc are set in the `getRequestBody()`, so `Metadata`  should be set here as well and minimize changes when future WI is worked on. So, even though we can set `Metadata` in various methods, it is better to leave it where it will be needed for the campaign code. Let me know if I overlooked something.

Reply:
Yes, that makes sense.